vmm_core/virt_mshv/src/lib.rs

```diff
@@ -1141,22 +1168,23 @@ impl virt::PartitionMemoryMap for MshvPartitionInner {
         }
         let slot_to_use = slot_to_use.unwrap();
 
-        let mut flags = HV_MAP_GPA_READABLE;
+        let mut flags = 0;
         if writable {
-            flags |= HV_MAP_GPA_WRITABLE;
+            flags |= set_bits!(u8, MSHV_SET_MEM_BIT_WRITABLE);
```

Comment:
Note: I have changed which flags are being used for user memory, as the "map_user_memory" from mshv-ioctl (which we call below at line 1187) is clearly expecting these flags. See https://github.com/rust-vmm/mshv/blob/c7edf4d5a2d73cacadba160b11c63098dc3696fd/mshv-ioctls/src/ioctls/vm.rs#L201